src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -479,8 +477,8 @@ def set_operator_statement(left: str, right: str, operator: str) -> str:
 def left_semi_or_anti_join_statement(
     left: str, right: str, join_type: JoinType, condition: str
 ) -> str:
-    left_alias = random_name_for_temp_object(TempObjectType.TABLE)
-    right_alias = random_name_for_temp_object(TempObjectType.TABLE)
+    left_alias = "SNOWPARK_TEMP_LEFT_RESULT"
```

Comment:
Do we need to worry about potential ambiguous alias if multiple joins are involved?